artifacts.go

```diff
@@ -97,6 +97,9 @@ type ArtifactConfig struct {
 	// Name is file or dir name to use for the artifact in the package.
 	// If empty, the file or dir name from the produced artifact will be used.
 	Name string `yaml:"name,omitempty" json:"name,omitempty"`
+	// PackageName is the name of the package. This can be used to replace
+	// `spec.Name` when destination paths are generated.
+	PackageName string `yaml:"packageName,omitempty" json:"packageName,omitempty"`
```

Comment:
My reasoning for suggesting a bool here is because the package name can go into `SubPath` and we'd just skip including the spec's package name with that.

Reply:
Next time I will clarify what you want before I implement the solution. I thought you meant that either was acceptable

Reply:
tldr: I think we should just use subPath and not do anything automatically. With an empty `subPath`, things go directly into `/usr/libexec` (which is standard). Otherwise, you can set `subPath` to `docker` if you want things to go into `/usr/libexec/docker` or `docker/cli-plugins` if you want `/usr/libexec/docker/cli-plugins`. Detailed explanation follows.

> My reasoning for suggesting a bool here is because the package name can go into `SubPath` and we'd just skip including the spec's package name with that.

So then in order to get the desired behavior in the example I've been using, you would need

```yaml
name: moby-buildx

artifacts:
  libexec:
    bin/docker-buildx:
      subPath: docker/cli-plugins
      omitAutoPackageName: true 
```

What I don't like about that is it kind of overloads the `subPath` field. When we don't need to change the package name, it would be

```yaml
name: docker

artifacts:
  libexec:
    bin/docker-buildx:
      subPath: cli-plugins
```

Both with the bool and without, it's a leaky abstraction. The user has to know several things:
1. the package name is inserted some times but not others (yes in /usr/lib artifacts, no in /usr/bin artifacts)
1. libexec will be one of the times when the name is inserted
1. in order to avoid inserting the name you need to set the bool
1. the bool changes the meaning of the `subPath` field, and changes the absolute path depth.

It's too much to ask of the user IMO. Better to just have `/usr/libexec` be the default (since that's a pretty standard thing to do) and have the user add the application name to the subpath if they want things nested in a subdirectory..